code-dot-org/code-dot-org
# Put the new survey pipline into service

6th PR in the new survey pipeline series. Putting it all together and ready to serve requests.

[Task PLC-47](https://codedotorg.atlassian.net/browse/PLC-47) -> [1st PR](https://github.com/code-dot-org/code-dot-org/pull/28736) -> [2nd PR](https://github.com/code-dot-org/code-dot-org/pull/28766) -> [3rd PR](https://github.com/code-dot-org/code-dot-org/pull/28770) -> [4th PR](https://github.com/code-dot-org/code-dot-org/pull/28816) -> [5th PR](https://github.com/code-dot-org/code-dot-org/pull/28823) -> [6th](https://github.com/code-dot-org/code-dot-org/pull/28846)

TO DO
- [x] Create a form table to look up form name.

**Example:** 
Requesting survey report for workshop id 6424.
(The tab currently shows form Id; it will show form's short name after form table is created in the 5th PR.)
![Screen Shot 2019-05-30 at 4 58 02 PM](https://user-images.githubusercontent.com/46507039/58672511-1eb41a80-82fc-11e9-8dda-4b9d3a1d5c9a.png)


**How tested:**
- Unit test: 
   - controller-only: `bin/rails test test/controllers/api/v1/pd/workshop_survey_report_controller_test.rb`
   - All new-pipeline tests: `bundle exec rails test test/lib/pd/survey_pipeline/daily_survey_retriever_test.rb test/lib/pd/survey_pipeline/daily_survey_parser_test.rb test/lib/pd/survey_pipeline/daily_survey_joiner_test.rb test/lib/pd/survey_pipeline/mapper_test.rb test/lib/pd/survey_pipeline/reducer_test.rb test/lib/pd/survey_pipeline/daily_survey_decorator_test.rb test/controllers/api/v1/pd/workshop_survey_report_controller_test.rb`

- Manual test:
  - Download survey question and a few submissions from real workshop (id 6424).
  - Create a CSF201 workshop in local environment and change its id to 6424 to match workshop id in the survey responds.
  - Test data-only request `http://localhost-studio.code.org:3000/api/v1/pd/workshops/6424/generic_survey_report`. It should return a JSON with necessary information about questions and summary results.
  - Test dashboard view request `http://localhost-studio.code.org:3000/pd/workshop_dashboard/daily_survey_results/6424`. ([This](https://studio.code.org/pd/workshop_dashboard/daily_survey_results/6424) is what it currently looks like in production for comparison.)
